src/wx/dialogs/speedup-config.cpp

```diff
@@ -0,0 +1,133 @@
+#include "wx/dialogs/speedup-config.h"
+
+#include <wx/checkbox.h>
+#include <wx/spinctrl.h>
+#include <wx/xrc/xmlres.h>
+
+#include "core/base/check.h"
+#include "wx/config/option-proxy.h"
+#include "wx/dialogs/base-dialog.h"
+
+#include "wx/wxlogdebug.h"
+
+namespace dialogs {
+
+// static
+SpeedupConfig* SpeedupConfig::NewInstance(wxWindow* parent) {
+    VBAM_CHECK(parent);
+    return new SpeedupConfig(parent);
+}
+
+SpeedupConfig::SpeedupConfig(wxWindow* parent)
+    : BaseDialog(parent, "SpeedupConfig") {
+
+    this->Bind(wxEVT_CHECKBOX, std::bind(&SpeedupConfig::ToggleSpeedupFrameSkip, this), XRCID("SpeedupThrottleFrameSkip"));
+    this->Bind(wxEVT_CHECKBOX, std::bind(&SpeedupConfig::ToggleSpeedupMute,      this), XRCID("SpeedupMute"));
+
+    GetValidatedChild<wxCheckBox>("SpeedupMute")->SetValue(OPTION(kPrefSpeedupMute));
+
+    using namespace std::placeholders;
+
+    this->Bind(wxEVT_SPIN_UP,   std::bind(&SpeedupConfig::SetSpeedupThrottle, this, _1), XRCID("SpeedupThrottleSpin"));
+    this->Bind(wxEVT_SPIN_DOWN, std::bind(&SpeedupConfig::SetSpeedupThrottle, this, _1), XRCID("SpeedupThrottleSpin"));
+    this->Bind(wxEVT_SPIN,      std::bind(&SpeedupConfig::SetSpeedupThrottle, this, _1), XRCID("SpeedupThrottleSpin"));
+    this->Bind(wxEVT_TEXT,      std::bind(&SpeedupConfig::SetSpeedupThrottle, this, _1), XRCID("SpeedupThrottleSpin"));
+
+    wxSpinCtrl* speedup_throttle_spin = GetValidatedChild<wxSpinCtrl>("SpeedupThrottleSpin");
+    wxCheckBox* frame_skip_cb         = GetValidatedChild<wxCheckBox>("SpeedupThrottleFrameSkip");
+
+    if (OPTION(kPrefSpeedupFrameSkip) != 0) {
+        speedup_throttle_spin->SetValue(OPTION(kPrefSpeedupFrameSkip) * 100);
+        prev_throttle_spin_ = OPTION(kPrefSpeedupFrameSkip) * 100;
+
+        frame_skip_cb->SetValue(true);
+        frame_skip_cb->Disable();
+    }
+    else {
+        speedup_throttle_spin->SetValue(OPTION(kPrefSpeedupThrottle));
+        prev_throttle_spin_ = OPTION(kPrefSpeedupThrottle);
+
+        frame_skip_cb->SetValue(OPTION(kPrefSpeedupThrottleFrameSkip));
+
+        if (OPTION(kPrefSpeedupThrottle) != 0)
+            frame_skip_cb->Enable();
+        else
+            frame_skip_cb->Disable();
+    }
+
+    speedup_throttle_spin->SetWindowStyle(wxSP_ARROW_KEYS);
+
+#ifdef wxTE_PROCESS_ENTER
+    speedup_throttle_spin->SetWindowStyleFlag(wxTE_PROCESS_ENTER);
+#endif
+
+    speedup_throttle_spin->Refresh();
+
+    this->Fit();
+    this->Refresh();
+}
+
+void SpeedupConfig::SetSpeedupThrottle(wxCommandEvent& evt)
```

Comment:
I would convert this to a validator but it can wait. The issue here is that the changes are going to take effect immediately, before the user hits "OK"